protocol/river-layout-v1.xml

```diff
@@ -0,0 +1,220 @@
+    phase. Backward compatible changes may be added together with the
+    corresponding interface version bump. Backward incompatible changes can
+    only be done by creating a new major version of the extension.
+  </description>
+
+  <interface name="river_layout_manager_v1" version="1">
+    <description summary="manage river layout objects">
+      A global factory for river_layout objects.
+    </description>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the river_layout_manager object">
+        This request indicates that the client will not use the
+        river_layout_manager object any more. Objects that have been created
+        through this instance are not affected.
+      </description>
+    </request>
+
+    <request name="get_river_layout">
+      <description summary="create a river_layout object">
+        This creates a new river_layout object for the given wl_output.
+
+        All layout related communication is done through this interface.
+
+        The namespace is used by the compositor to decide which river_layout
+        object will receive layout demands for the output. Namespaces must be
+        unique per output. The same namespace may occur on other outputs, as
+        long as that river_layout object comes from the same client. If the
+        namespace is found to not obey these uniqueness rules, the server will
+        send a namespace_in_use event on the river_layout object immediately
+        after creation and ignore all requests on the object aside from the
+        destroy request.
+      </description>
+      <arg name="id" type="new_id" interface="river_layout_v1"/>
+      <arg name="output" type="object" interface="wl_output"/>
+      <arg name="namespace" type="string" summary="namespace of the layout object"/>
+    </request>
+  </interface>
+
+  <interface name="river_layout_v1" version="1">
+    <description summary="receive and respond to layout demands">
+      This interface allows clients to receive layout demands from the
+      compositor for a specific output and subsequently propose positions and
+      dimensions of individual views.
+    </description>
+
+    <enum name="error">
+      <entry name="proposed_dimension_mismatch" value="0" summary="amount of
+        proposed dimensions does not match amount of views in layout"/>
+    </enum>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the river_layout object">
+        This request indicates that the client will not use the river_layout
+        object any more.
+      </description>
+    </request>
+
+    <event name="namespace_in_use">
+      <description summary="namespace already in use">
+        The requested namespace is already used by another river_layout object.
+        After receiving this event, the client should destroy the river_layout
+        object. Any other request will be ignored.
+      </description>
+    </event>
+
```

Comment:
Why this is a separate event instead of an error?

Reply:
errors are fatal, and the client can't possible know that the namespace was already in use

Reply:
But errors aren't so fatal that they cause clients to drop, are they?

Reply:
```
12:53 <yyp> https://github.com/ifreund/river/pull/112#discussion_r615375314 - do I not understand Wayland errors? I thought they are just a different response to a request indicating that something went wrong, do they do anything more?
12:57 <ifreund> yes, they disconnect the client
13:02 <yyp> Ok, makes sense. But do we really need to keep the layout client connected when it's namespace is taken?
13:02 <yyp> s/it's/its
13:04 <ifreund> we need to allow clients to try again with a different namespace or report an error as they see fit
13:05 <ifreund> wayland errors should only be used for protocol violations
13:05 <ifreund> and protocol violations should never happen in a bug free client/compositor
```

Reply:
> But do we really need to keep the layout client connected when it's namespace is taken?

Yes. A client could very well have multiple layouts are some other functionality important to the user. We should not interrupt that just because some other client was faster at registering  a layout with a specific name.